### ISSUE TITLE
filenames in releases

### ISSUE THREAD

**Issue by Alexey-T** [davidbannon/libqt6pas#5]
libqt6pas-dev-2.2-1.x86_64.rpm
libqt6pas-dev_2_2-1_amd64.deb
libqt6pas6-2.2-1.x86_64.rpm
libqt6pas6_2_2-1_amd64.deb
libqt6pas_6_2_2-1_amd64.tar.gz 

Usually I read recommendations to name DEB packages like this:
cudatext_1.177.0.0-1_gtk2_amd64.deb

Here you see version number DOT separated, and version is separated from name by underscore, and from widgetset name , and from CPU.

@davidbannon

**Comment by davidbannon** [comment]
Alexey, as I understand it, Debian does not define a widget set name in the deb file name. After all, its not that common to have an environment that build apps and easily switch between widget sets. As the version number appears last in the main part of the file name, having eg gtk2 just before the ".deb" will, I suspect confuse some parsing systems. But probably does not matter. The version number would be seen as "1.177.0.0-1_gtk2" and no packaging number perhaps ? "1.177.0.0-1_qt6" would be a later version number because 'q' is higher than 'g'.

In my view, cudatext-gtk2_1.177.0.0-1_amd64.deb would be a better model. The particular package is, in this case, the gtk2 version of cudatext, so it should be known as cudatext-gtk2. 

_This from memory, don't treat as authoritative -
If you have, for example two packages, the gtk2 and qt5 versions of cudatext, same version number and both provide, eg, the cudatext binary (with the Provides: tag), then installing one on top of the other (with apt) will not work, you must remove one and then reinstall the other. (Interestingly, using gdebi it does work, apparently its smart enough to note both provide the same binary so just go off and do it.)_

The underscore is an acceptable separator in version number stanzas, in this case, it was an attempt to deal with the way Qt appends a '6' to the main package name even though its included in the main name already. Note its not the case on the dev package. I don't think its necessary to use the underscore any more as we now have a predictable and sequential version number. 

As the scripts are being somewhat reworked at present, (and alien being removed) I'll see if I can revert back to dots again in the next release/repackage. But there may be problems the '_' being greater than the '.' .....

Like we used to have with the '~' being part of the old libqt6pas number !

Davo

**Comment by Alexey-T** [comment]
> "1.177.0.0-1_qt6" would be a later version number because 'q' is higher than 'g'.

no, it is still 1.177.0.0-1 because version is read from meta file (included). filename is only nice to have.

**Comment by Alexey-T** [comment]
No problem if you name package with double 6 - libqt6_6.2.2.0-1_amd64.xxx

**Comment by davidbannon** [comment]
> No problem if you name package with double 6 - libqt6pas_6.2.2.0-1_amd64.xxx

Not sure what you mean b that Alexey, is it a comment on the header of the script I pushed up yesterday >

https://github.com/davidbannon/libqt6pas/blob/master/cbindings/package/package-lib

In fact, many packages in the debian repo have the '6' three times !  I think its strange, silly and unnecessary but I am now altering my script to do exactly that !  So the deb packages will look like this -

libqt6pas6_6.2.0-1_amd64.deb  and libqt6pas6-dev.6.2.0-1_amd64.deb

This is quite different from previous ways of doing it but is necessary to avoid a rpmlint error message.

Incidentally, the number 666 does have some significance, google !   :-0

Davo

**Comment by Alexey-T** [comment]
if the number occured in your name (somehow coded) in the middle ages, you were doomed.

**Comment by davidbannon** [comment]
Indeed ! I hope we are not facing a similar situation here. 

I have now uploaded new packages in a new package name format. Sadly, it requires existing users to remove and replace. But no need to do so if what they have now works of course. 

You will like the new format because it has dots in the version number, I like it because I can now build (relatively) error free rpms (with out using alien, a big improvement). I found a number of problems in the way alien was making rpms, not good. If you make rpms for cudatext with alien, I suggest you have a look through package-lib

This is a tool to check rpms much like debian's lintian called rpmlint but its buggy and not as well documented. 

Davo
